src/resources/HostedPages/HostedPages.model.ts

```diff
@@ -0,0 +1,87 @@
+import {Paginated} from '../BaseInterfaces.js';
+
+export interface ListHostedPagesParams extends Paginated {
+    /**
+     * A substring that must appear in a search interface configuration name for this configuration to appear in results.
+     */
+    filter?: string;
+
+    /**
+     * The sort direction of the results.
+     * Possible values:
+     * - `'asc'`: ascending order
+     * - `'desc'`: descending order.
+     */
+    order?: 'desc' | 'asc';
+}
+
+export interface HostedPage {
+    /**
+     * The unique identifier of the hosted page.
+     */
+    id: string;
+    /**
+     * A short descriptive name to help manage hosted pages.
+     */
+    name: string;
+    /**
+     * The HTML markup of the hosted page.
+     */
+    html: string;
+    /**
+     * The JavaScript resources appended to the header of a specific hosted page in the target Coveo Cloud organization.
+     */
+    javascript?: HostedPageJavascriptFile[];
+    /**
+     * The CSS resources appended in the header, pertaining to a specific hosted page in the target Coveo Cloud organization.
+     */
+    css?: HostedPageCssFile[];
+}
+
+export interface HostedPageCssFile {
+    /**
+     * The content of the header `style` tag. If this property is defined, the `url` property should not be specified.
+     */
+    inlineContent?: string;
+    /**
+     * The URL of CSS stylesheet. If this property is defined, the `inlineContent` property should not be specified.
+     */
+    url?: string;
+}
```

Comment:
In cases like this one where it can only be A or B, I think we should use union types.

```suggestion
export interface HostedPageInlineCssFile {
    /**
     * The content of the header `style` tag.
     */
    inlineContent?: string;
}

export interface HostedPageUrlCssFile {
    /**
     * The URL of CSS stylesheet.
     */
    url?: string;
}

export type HostedPageCssFile = HostedPageInlineCssFile | HostedPageUrlCssFile;
```

Reply:
Great!